lib/prawn/font.rb

```diff
@@ -68,6 +71,48 @@ def font(name = nil, options = DEFAULT_OPTS)
       @font
     end
 
+    def font_family
+      @font ? @font.family : 'Helvetica'
+    end
+
+    # @method font_style(points=nil)
+    #
+    # When called with no argument, returns the current font style.
+    #
+    # When called with a single argument but no block, sets the current font
+    # style.  When a block is used, the font style is applied transactionally and
+    # is rolled back when the block exits.  You may still change the font style
+    # within a transactional block for individual text segments, or nested calls
+    # to font_style.
+    #
+    #   Prawn::Document.generate("font_style.pdf") do
+    #     font_style :bold
+    #     text "Bold text"
+    #
+    #     font_style(:italic) do
+    #       text "Italic text"
+    #       text "Bold text", :style => :bold
+    #       text "Italic text"
+    #     end
+    #
+    #     text "Bold text"
+    #   end
+    #
+    # Font styles are not additive, i.e. if the current style is :bold, setting
+    # the style to :italic results in italic, not bold-italic text. Use
+    # :bold_italic instead.
+    #
+    def font_style(style = nil, &block)
+      return @font ? @font.style : :normal if style.nil?
+
+      font(font_family, style: style, &block)
```

Comment:
So, thinking a bit more about this, why not just pass `nil` here? Then it would automatically use what's defined in `#font` and we would have only one place where it would be defined.

Reply:
Another thing that came to mind: What happens if we pass the path to a e.g. TrueType font into `#font` and then call `font_style(:bold)`?

Since nothing is defined and we cannot know which file contains the bold version, I guess the only thing to do is to error out here. What do you think?

Reply:
Maybe we should check here for `font_families.key?(font_family)` and error out if nothing is found. This would solve both problems, the one when using a TrueType font path and the one where the used key is different from the family name. The error message should probably include the reason for the failure, i.e. that the family name was not registered.

And updating the documentation for `#font_families` in this regard would also make sense.

Reply:
I agree that calling `font_style` with a key not present in `font_families` for the current font should raise an error. Likewise if style is other than `normal` for a specific font file.

Currently `font` does not raise in the latter situation, and changing this would be a BC break, so I suggest we avoid raising in that specific situation (`font` called with style other than normal for a specific font file). I imagine that it is not uncommon that people do e.g. `font 'ArialBlack.ttf', style: :bold` in some (mis-guided) attempt to signal that this is a bold font.

Reply:
Agreed on both accounts.

@pointlessone What do you think?

Reply:
It's a little confusing because we have two very similar things:
- `font_families` which afford for styles
- `font_registry` which does not quite do that.

`font_families` eventually gets mapped to entries in `font_registry`. Style is only meaningful on the stage of this mapping. Whenever a font file is used directly style has no meaning. It still will be present in the font but there's no way to look up a font for any other style. We never add these fonts to `font_families`, we never look up fonts by their internal names so we're unable to correlate fonts from the same family. I don't think this should change now. Maybe in the next major release if we ever get there.

I suggest, we make `font_style(:random_style) { ... }` a noop but issue a `warn`ing. Possibly, when `$DEBUG` is set but it rarely is so it's not very useful.